flake.nix

```diff
@@ -15,151 +15,167 @@
-            created = "now";
-            config.Entrypoint = [ "${packages.prod}/bin/prod" ];
-            config.Cmd = [ "version" ];
-          };
-          oci = pkgs.ociTools.buildContainer {
-            args = [ "${packages.prod}/bin/prod" ];
-          };
-          default = packages.prod;
-        };
-
-        # apps to run with nix run
-        apps = {
-          prod = flake-utils.lib.mkApp {
-            name = pname;
-            drv = packages.prod;
-            exePath = "/bin/prod";
+    flake-utils.lib.eachSystem flake-utils.lib.defaultSystems
+      (system:
+        let
+          inherit (nixpkgs.lib) optional;
+          pkgs = import nixpkgs { inherit system; };
+
+          # project name for mix release
+          pname = props.app_name;
+          # project version for mix release
+          version = props.app_version;
+
+          # use ~r/erlangR[1-9]+/ for specific erlang release version
+          beamPackages = pkgs.beam.packagesWith
+            pkgs.beam.interpreters.${erlang_nix_version props.erlang_release};
+          # all elixir and erlange packages
+          erlang = beamPackages.erlang;
+          # use ~r/elixir_1_[1-9]+/ major elixir version
+          elixir = beamPackages.${elixir_nix_version props.elixir_release};
+          elixir-ls = beamPackages.elixir_ls.overrideAttrs
+            (oldAttrs: rec { elixir = elixir; });
+          hex = beamPackages.hex;
+
+          # use rebar from nix instead of fetch externally
+          rebar3 = beamPackages.rebar3;
+          rebar = beamPackages.rebar;
+
+          # This is setting locality to latin for some reason
+          locality = pkgs.glibcLocales;
+
+          # Install hook for mixRelease 
+          installHook = ''
+            export APP_VERSION="${version}"
+            export APP_NAME="${pname}"
+            export ELIXIR_RELEASE="${props.elixir_release}"
+            runHook preInstall
+            mix release --no-deps-check --path "$out"
+            runHook postInstall
+          '';
+
+          # Pass just as a build input 
+          inputsBuild = with pkgs; [ just ];
+
+          # Had to change the find command because it wasn't working
+          fetchMixInstallPhase = { mixEnv }: ''
+            runHook preInstall
+            mix deps.get --only ${mixEnv}
+            find "$TEMPDIR/deps" -path '*/.git/*' -a ! -name HEAD -prune -execdir rm -rf {} + 
+            cp -r --no-preserve=mode,ownership,timestamps $TEMPDIR/deps $out
+            runHook postInstall
+          '';
+
+          # Run just before compiling to set up the environment
+          configureHook = ''
+            runHook preConfigure
+            ${./mix-configure-hook.sh}
+
+            # Run just and setup config directory before we compile 
+            # TODO allow choosing flavor
+            just rel-init
+            just rel-prepare
+            just assets-prepare
+            cp -r data/current_flavour/config/ ./config/
```

Comment:
are these the right commands to be running? If so see the changes necessary for the [justfile](https://github.com/bonfire-networks/bonfire-app/pull/459/files#diff-deb9bb56fb122db0b605aa5b63f95a4665c905b18dd670e1fa6c877576a94ff1R94)